### ISSUE TITLE
Custom icon upload does not work

### ISSUE THREAD

**Issue by Freekers** [domoticz/domoticz#2925]
When trying to upload custom icons, *nothing* happens if you press the 'Upload' button.
In the Chrome console, the following error is displayed: 
```
TypeError: $http.post(...).success is not a function
    at b.$scope.uploadFileToUrl (CustomIconsController.js:14)
    at b.$scope.UploadIconSet (CustomIconsController.js:34)
    at fn (eval at compile (angular.min.js:242), <anonymous>:4:159)
    at e (angular.min.js:288)
    at b.$eval (angular.min.js:151)
    at b.$apply (angular.min.js:151)
    at HTMLButtonElement.<anonymous> (angular.min.js:288)
    at HTMLButtonElement.dispatch (jquery-3.3.1.min.js:2)
    at HTMLButtonElement.y.handle (jquery-3.3.1.min.js:2`)
```
The error is caused by line 14 in file domoticz\www\app\CustomIconsController.js 
In Angular 1.6, the $http service has changed: you can't use .success and .error anymore. Instead, use .then. (From $http docs)

The error can be fixed as follows. Replace this:
```
.success(function (data) {
	if (data.status != "OK") {
		HideNotify();
		ShowNotify($.t('Error uploading Iconset') + ": " + data.error, 5000, true);
	}
	$scope.RefreshIconList();
})
.error(function (data) {
	HideNotify();
	ShowNotify($.t('Error uploading Iconset'), 5000, true);
});
```

By this:
```
.then(function(response) {
	//Success
	var data = response.data;
	if (data.status != "OK") {
		HideNotify();
		ShowNotify($.t('Error uploading Iconset') + ": " + data.error, 5000, true);
	}
	$scope.RefreshIconList();
}, function(response) {
	//Error
	HideNotify();
	ShowNotify($.t('Error uploading Iconset'), 5000, true);
}); 
```

I'm running stable 4.9700, but according to the source code here on Github, this issue is still present in the latest Development branch.

Thank you!

Source: https://www.domoticz.com/forum/viewtopic.php?t=24201

**Comment by gizmocuz** [comment]
Thanks for reporting. If you like you can make a pull request next time... just read your fix... after mine *g*

**Comment by Freekers** [comment]
Thanks for the quick fix. Noted, will do next time!

Happy holidays

**Comment by kitopopo** [comment]
Hi dear friend @gizmocuz ,

I get an error loading the icon collection. undefined. I have reinstalled domoticz and without loading any database the problem persists, there is no way to upload icons always the same error. I have also given write permissions to the entire domoticz folder on a raspberry 4. 

in the domoticz forum people are experiencing problems again

`https://www.domoticz.com/forum/viewtopic.php?f=28&t=30754&p=249186#p249186`

I have also tried several icon packs and the problem persists. Is there a solution? I have the latest beta version of domoticz installed. Thanks in advanced



Version: 2020.2 (build 12106)
Build Hash: b08d17c2d-modified
Compile Date: 2020-05-29 09:01:13
dzVents Version: 3.0.8
Python Version: 3.7.3 (default, Dec 20 2019, 18:57:59) [GCC 8.3.0
